src/components/cadastro/index.tsx

```diff
@@ -0,0 +1,128 @@
+import React, { useState } from 'react';
+import './index.css';
+
+function Cadastro() {
+  const [mostraError, setMostraError] = useState(false);
+  const [erros, setErros] = useState<string[]>([]);
+  const [email, setEmail] = useState('');
+  const [senha, setSenha] = useState('');
+  const [telefone, setTelefone] = useState('');
+  const [dataNascimento, setDataNascimento] = useState('');
+  const [nomeCompleto, setNomeCompleto] = useState('');
+
+  function handleSubmit(event: React.FormEvent<HTMLFormElement>) {
+    event.preventDefault();
+
+    function validacao(email: string, senha: string, telefone: string, DataNascimento: string, nomeCompleto: string): string[] {
+      const erros: string[] = [];
+
+      if (!email) {
+        erros.push('O e-mail é obrigatório.');
+      } else if (!/\S+@\S+\.\S+/.test(email)) {
+        erros.push('E-mail inválido.');
+      }
+
+      if (!senha) {
+        erros.push('A senha é obrigatória.');
+      } else if (senha.length < 7) {
+        erros.push('A senha deve ter pelo menos 7 caracteres.');
+      } else if (!/[a-zA-Z]/.test(senha) || !/\d/.test(senha)) {
+        erros.push('A senha deve ter pelo menos uma letra e um dígito.');
+      }
+      if (!telefone) {
+        erros.push('O telefone é obrigatório.');
+      } else if (!/^\(\d{2}\)\s\d{4,5}-\d{4}$/.test(telefone)) {
+        erros.push('Telefone inválido. O formato deve ser (XX) XXXXX-XXXX.');
+      }
+
+      if (!dataNascimento) {
+        erros.push('A data de nascimento é obrigatória.');
+      } else {
+        const hoje = new Date();
+        const data = new Date(dataNascimento);
+        let idade = hoje.getFullYear() - data.getFullYear();
```

Comment:
para o calculo direitinho da idade, está errado....ai na validação de ter pelo menos 15 anos ali embaixo pode ter caso que de 15 mas a pessoa ainda não fez aniversário, então se a idade realmente for importante e vc deixar passar alguem que não poderia pode dar um baita problema